pkg/operator/controller.go

```diff
@@ -777,25 +775,36 @@ func (c *Controller) getProviderName(nodeName string) (string, error) {
 	return util.GetProviderName(node.Spec.ProviderID), nil
 }
 
-func (c *Controller) syncMachineConfig(pools []*mcfgv1.MachineConfigPool, labels map[string]string, profile *tunedv1.Profile) error {
+func (c *Controller) syncMachineConfig(labels map[string]string, profile *tunedv1.Profile) error {
 	var (
+		bootcmdline     string
 		kernelArguments []string
 	)
 
-	name := GetMachineConfigNameForPools(pools)
-	nodes, err := c.pc.getNodesForPool(pools[0])
+	pools, err := c.pc.getPoolsForMachineConfigLabels(labels)
 	if err != nil {
 		return err
 	}
 
-	bootcmdline, bootcmdlineSet := c.pc.state.bootcmdline[profile.Name]
-	if !bootcmdlineSet {
-		klog.V(2).Infof("syncMachineConfig(): bootcmdline for %s not cached, sync canceled", profile.Name)
-		return nil
-	}
+	name := GetMachineConfigNameForPools(pools)
+	klog.V(2).Infof("syncMachineConfig(): %v", name)
 
-	if ok := c.allNodesAgreeOnBootcmdline(nodes); !ok {
-		return fmt.Errorf("not all %d Nodes in MCP %v agree on bootcmdline: %s", len(nodes), pools[0].ObjectMeta.Name, bootcmdline)
+	for _, pool := range pools {
+		var bootcmdlineSet bool
+		nodes, err := c.pc.getNodesForPool(pool)
+		if err != nil {
+			return err
+		}
+
+		bootcmdline, bootcmdlineSet = c.pc.state.bootcmdline[profile.Name]
```

Comment:
could state be nil ?

Reply:
No. And we've been running with that code for a long time.